### ISSUE TITLE
TypeError: 'NoneType' object is not iterable (Viomi S9)

### ISSUE THREAD

**Issue by fokcuk** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#158]
After the last HA update (or could be few others - did not check) to 2021.7.4 started to see a lot of these in the logs

Logger: homeassistant.helpers.entity
Source: custom_components/xiaomi_cloud_map_extractor/roidmi/map_data_parser.py:117
Integration: xiaomi_cloud_map_extractor (documentation, issues)
First occurred: 11:36:41 PM (35 occurrences)
Last logged: 11:39:32 PM

Update for camera.xiaomi_cloud_map_extractor fails
Traceback (most recent call last):
```
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 401, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 609, in async_device_update
    raise exc
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/xiaomi_cloud_map_extractor/camera.py", line 257, in update
    self._handle_map_data(map_name)
  File "/config/custom_components/xiaomi_cloud_map_extractor/camera.py", line 314, in _handle_map_data
    map_data, map_stored = self._device.get_map(map_name, self._colors, self._drawables, self._texts,
  File "/config/custom_components/xiaomi_cloud_map_extractor/common/vacuum.py", line 28, in get_map
    map_data = self.decode_map(response, colors, drawables, texts, sizes, image_config)
  File "/config/custom_components/xiaomi_cloud_map_extractor/roidmi/vacuum.py", line 15, in decode_map
    return MapDataParserRoidmi.parse(unzipped, colors, drawables, texts, sizes, image_config)
  File "/config/custom_components/xiaomi_cloud_map_extractor/roidmi/map_data_parser.py", line 31, in parse
    map_data.rooms = MapDataParserRoidmi.parse_rooms(map_info)
  File "/config/custom_components/xiaomi_cloud_map_extractor/roidmi/map_data_parser.py", line 117, in parse_rooms
    for area in areas:
TypeError: 'NoneType' object is not iterable
```

**Comment by r3pek** [comment]
Any idea what this is? Just got the same problem :(

**Comment by PiotrMachowski** [comment]
Problem should be solved in v2.1.2

**Comment by TriptonPt** [comment]
I am getting the same issue on Home Assistant 2022.9.4 with latest version 2.2.0 

Vacuum Model : Viomi S9

@PiotrMachowski any fix ? 

Thanks !
